src/thread_task.rs

```diff
@@ -935,24 +914,25 @@ pub unsafe fn rav1d_worker_task(c: &Rav1dContext, task_thread: Arc<Rav1dTaskCont
                                 // TODO(sjc): t is a reference to an array
                                 // element, we need to replace this pointer
                                 // arithmetic with proper indexing
-                                let next_t: *mut Rav1dTask =
-                                    &mut *(t as *mut Rav1dTask).offset(1) as *mut Rav1dTask;
-                                *next_t = t.clone();
-                                (*next_t).sby += 1;
+                                let next_t_idx = t_idx + 1;
+                                let t = t.clone();
```

Comment:
Sure, but we should not be asserting a particular size here. We don't want to update this assert or asserts if we shrink the Task size, for example. Asserts, even compile-time ones, are for correctness, not performance.

Reply:
My point was that we do want a breakage here if the clone suddenly got large.  Then we could update or modify/remove the assert if our calculus changed.  You said this clone should be fine because it's small, so I think we should make that assumption explicit.  It's good for readibility and avoiding unexpected changes at a distance.

Reply:
If we grow the structure too much, we should remove Clone and this will break anyway. This isn't a place we want to incur unintentional breakage. Performance here is relative, and we don't know how large this can be without incurring a cost, so I don't have any hard limit we can impose here.

Reply:
Okay, could we at least just leave a comment, though?